src/Core/Settings.h

```diff
@@ -446,6 +446,7 @@ class IColumn;
     M(Bool, optimize_read_in_order, true, "Enable ORDER BY optimization for reading data in corresponding order in MergeTree tables.", 0) \
     M(Bool, optimize_read_in_window_order, true, "Enable ORDER BY optimization in window clause for reading data in corresponding order in MergeTree tables.", 0) \
     M(Bool, optimize_aggregation_in_order, false, "Enable GROUP BY optimization for aggregating data in corresponding order in MergeTree tables.", 0) \
+    M(Bool, optimize_join_in_order, true, "Enable JOIN optimization for joining data using `full_sorting_merge` algorithm in corresponding order in MergeTree tables.", 0) \
```

Comment:
Please rename to `query_plan_joing_in_order` and move near `query_plan_read_in_order`

Reply:
These optimizations are based on AST (old versions). Current versions of reading in order optimizations are done on top of the plan and named similarly but started with `query_plan_...`